### DISCUSSION TITLE
API [POST] camt_converter_ISO20022_for_camt

### DISCUSSION THREAD

**Discussion by plumped** [firefly-iii/firefly-iii#7488] [Firefly III]
@JC5  can you please give me an advise how a pattern has to look like for the transaction creation on the swagger api. I already read the swagger documentation, but im not sure what exactly i have to transfer for a creation. Now that my code is optimized and ready to run, I'll take on the API interface. Thanks in advance for a template

**Comment by JC5** [comment]
Following [this API endpoint](https://api-docs.firefly-iii.org/#/transactions/storeTransaction), the bare minimum you must submit is the following:

```json
{
    "transactions": [
        {
            "type": "withdrawal",
            "source_id": 1,
            "description": "Some description",
            "amount": "10",
            "date": "2022-03-31"
        }
    ]
}
```

You must determine yourself if a transaction is a `withdrawal`, `deposit` or `transfer`.

The `source_id`, `source_iban`, `source_number`, `source_name` and similar `destination_*` fields will be used to determine which accounts are the source account and destination account respectively. There are some rules for these fields:

* Withdrawals MUST have a source that exists and is an asset account. You can refer to this asset account by ID, name, IBAN or account number (any one is fine). In my example, I simply refer to asset account 1.
* Same for deposits, but then the destination account.
* Same for transfers, but for BOTH accounts.
* Liabilities also work in these cases.
* For `withdrawal` transactions the destination account must be an expense account. You can refer to this account by ID, name, IBAN or account number (any one is fine). If you refer to this account by name, IBAN or account number *and it does not yet exist* it will be automatically created. 
* For `deposit` transactions, this applies to the source account.
* In either case, asset accounts can never be creaed this way.

The rest of the fields you can get from the API docs. Categories and tags will be automatically created if they do not exist, budgets will not.
